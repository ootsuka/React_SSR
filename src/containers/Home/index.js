import React, { Component } from 'react'
import { connect } from 'react-redux'

import { getNewsList } from './store/actions'
import Header from '../../components/Header'

class home extends Component {
  componentDidMount() {
    if (!this.props.list.length) {
      this.props.getList()
    }
  }
  genList() {
    return (
      this.props.list.map((item) => <div key={item.id}>{item.title}</div>)
    )

  }
  render() {
    return(
      <div>
        <Header />
        <div>{this.genList()}</div>
        <button onClick={() => alert('clicked')}>
          click
        </button>
      </div>

    )
  }
}

home.loadData = (store) => {
  //load data before server renders
  return store.dispatch(getNewsList())
}
const mapStateToProps = state => ({
  name: state.home.name,
  list: state.home.newsList,
})
const mapDispatchToProps = dispatch => ({
  getList: () => dispatch(getNewsList())
})
export default connect(mapStateToProps, mapDispatchToProps)(home)
