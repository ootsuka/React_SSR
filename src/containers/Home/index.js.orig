import React, { Component, Fragment } from 'react'
import { connect } from 'react-redux'
import { Helmet } from 'react-helmet'

import { getNewsList } from './store/actions'
import styles from './style.css'
import withStyle from '../../withStyle'
<<<<<<< HEAD
<<<<<<< HEAD
import Footer from '../Footer/index'
=======
import SideBar from '../SideBar/index'
>>>>>>> 94de39019765a073dbe61268fef0610f46016f9c
=======
import SideBar from '../SideBar/index'
>>>>>>> 94de39019765a073dbe61268fef0610f46016f9c

class home extends Component {
  componentDidMount() {
    if (!this.props.list.length) {
      this.props.getList()
    }
  }

  genList() {
    return (
<<<<<<< HEAD
      this.props.list.map((item) => <div className={styles.item} key={item.id}>{item.title}</div>)
=======
      this.props.list.map((item) => <div key={item.id}>{item.title}</div>)
>>>>>>> 94de39019765a073dbe61268fef0610f46016f9c
    )

  }
  render() {
    return(
      <Fragment>
        <Helmet>
          <title>news feed</title>
          <meta name="description" content="this is a news list page" />
        </Helmet>
<<<<<<< HEAD
<<<<<<< HEAD
        <div className={styles.container}>
          <div>{this.genList()}</div>
          <Footer />
=======
=======
>>>>>>> 94de39019765a073dbe61268fef0610f46016f9c
        <div className={styles.test}>
          <div>{this.genList()}</div>
          <button onClick={() => this.handleClick()}>
            click
          </button>
          <SideBar />
<<<<<<< HEAD
>>>>>>> 94de39019765a073dbe61268fef0610f46016f9c
=======
>>>>>>> 94de39019765a073dbe61268fef0610f46016f9c
        </div>
      </Fragment>
    )
  }
}

const mapStateToProps = state => ({
  name: state.home.name,
  list: state.home.newsList,
})
const mapDispatchToProps = dispatch => ({
  getList: () => dispatch(getNewsList())
})
const exportHome = connect(mapStateToProps, mapDispatchToProps)(withStyle(home, styles))

exportHome.loadData = (store) => {
  //load data before server renders
  return store.dispatch(getNewsList())
}

export default exportHome
