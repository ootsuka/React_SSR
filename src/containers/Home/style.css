<<<<<<< HEAD
.container {
  margin-top: 20px;
  margin-left: 20px;
}
.item {
  line-height: 34px;
  font-size: 16px;
  color: #666;
=======
body {
  background: red
}
.test {
  background: blue
>>>>>>> 94de39019765a073dbe61268fef0610f46016f9c
}
