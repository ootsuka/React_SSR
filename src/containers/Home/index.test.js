import React from 'react'
import { shallow } from 'enzyme'

import Home from './index.js'

describe('Home', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Home />)

    expect(component).toMatchSnapshot()
  })
})
