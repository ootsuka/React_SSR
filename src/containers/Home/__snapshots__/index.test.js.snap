// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
